src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -0,0 +1,402 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonPath;
+
+use Symfony\Component\JsonPath\Exception\InvalidInputJsonException;
+use Symfony\Component\JsonPath\Exception\InvalidJsonPathException;
+use Symfony\Component\JsonPath\Exception\JsonCrawlerException;
+use Symfony\Component\JsonPath\Tokenizer\JsonPathToken;
+use Symfony\Component\JsonPath\Tokenizer\JsonPathTokenizer;
+use Symfony\Component\JsonPath\Tokenizer\JsonPathTokenReducer;
+use Symfony\Component\JsonPath\Tokenizer\TokenType;
+
+/**
+ * Crawls a JSON document using a JSON Path as described in the RFC 9535.
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc9535
+ *
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ *
+ * @experimental
+ */
+final class JsonCrawler implements JsonCrawlerInterface
+{
+    private const RFC9535_FUNCTIONS = [
+        'length' => true,
+        'count' => true,
+        'match' => true,
+        'search' => true,
+        'value' => true,
+    ];
+
+    public function __construct(
+        private readonly string $raw,
+    ) {
+    }
+
+    public function find(string|JsonPath $query): array
+    {
+        return $this->evaluate(\is_string($query) ? new JsonPath($query) : $query);
+    }
+
+    private function evaluate(JsonPath $query): array
+    {
+        try {
+            $simplified = JsonPathTokenReducer::findSmallestDeserializableStringAndPath(
+                JsonPathTokenizer::tokenize($query),
+                $this->raw,
+            );
```

Comment:
This will require `symfony/json-encoder`. I'd make this feature opt-in if the package is installed.
And if the `symfony/json-encoder` is not installed and `$this->raw` is a resource, I'll have thrown a meaningful exception telling to install the package.

Reply:
That was indeed missing, thanks!